SimulationModule/Models/KafkaPositionObserver.py

```diff
@@ -0,0 +1,28 @@
+from Models.IPositionObserver import IPositionObserver
+from Models.GpsSensor import GpsSensor
+from Models.PositionJsonAdapter import PositionJsonAdapter
+import threading
+from abc import ABC, abstractmethod
+
+class KafkaPositionObserver(IPositionObserver,ABC):
+    '''This class implements the observer interface and is used to observe the GPS sensor position and
+    write the position to a Kafka topic, note this class will be inherited by a KafkaConfluentAdapter'''
+    
+    def __init__(self, json_adapter_istance: PositionJsonAdapter):
```

Comment:
```suggestion
    def __init__(self, json_adapter_istance: IJsonSerializable):
```
Attento che qua sarebbe da mettere il nome dell'interfaccia e NON della classe implementata, sennò aver fatto l'interfaccia non servirebbe a nulla e non staresti rispettando l'associazione presente tra questa classe e l'interfaccia.

Reply:
yes, sistemo